src/dsymbolsem.c

```diff
@@ -4880,8 +4880,11 @@ static void aliasInstanceSemantic(TemplateInstance *tempinst, Scope *sc, Templat
 
     TemplateTypeParameter *ttp = (*tempdecl->parameters)[0]->isTemplateTypeParameter();
     Type *ta = isType(tempinst->tdtypes[0]);
-    Declaration *d = new AliasDeclaration(tempinst->loc, ttp->ident, ta->addMod(tempdecl->onemember->isAliasDeclaration()->type->mod));
-    d->storage_class |= STCtemplateparameter;
+    AliasDeclaration *ad = tempdecl->onemember->isAliasDeclaration();
+
+    // Note: qualifiers can be in both 'ad.type.mod' and 'ad.storage_class'
```

Comment:
How come this is the case at all?

Reply:
It's the way parsing alias works, it has to parse attributes first (const, immutable, etc. are also attributes) to support for example declaring function pointers. The alias semantic phase decides whether it's a type qualifier but is skipped for this trivial alias shortcut.